src/game.zig

```diff
@@ -1023,8 +1031,11 @@ pub fn update(deltaTime: f64) void { // MARK: update()
 					movementDir[2] -= walkingSpeed;
 				}
 			}
+
 			if(movementSpeed != 0 and vec.lengthSquare(movementDir) != 0) {
-				movementDir = vec.normalize(movementDir);
+				if(vec.lengthSquare(movementDir) > 1) {
```

Comment:
```suggestion
				if(vec.lengthSquare(movementDir) > movementSpeed*movementSpeed) {
```
movementDir is determined from components scaled by `movementSpeed`.
If it's smaller than that, then we need to divide it by `movementSpeed` to get a correct value.